pyvista/utilities/__init__.py

```diff
@@ -1,77 +1,39 @@
-"""Utilities routines."""
-from .algorithms import algorithm_to_mesh_handler, set_algorithm_input
-from .common import perlin_noise, sample_function
-from .errors import (
-    GPUInfo,
-    Observer,
-    Report,
-    assert_empty_kwargs,
-    get_gpu_info,
-    send_errors_to_logging,
-    set_error_output_file,
-    check_valid_vector,
-    VtkErrorCatcher,
-)
-from .features import *
-from .fileio import *
-from .geometric_objects import *
-from .helpers import *
-from .parametric_objects import *
-from .sphinx_gallery import Scraper, _get_sg_image_scraper
-from .regression import compare_images
-from . import transformations
-from .xvfb import start_xvfb
-from .reader import (
-    get_reader,
-    AVSucdReader,
-    BaseReader,
-    BinaryMarchingCubesReader,
-    BMPReader,
-    BYUReader,
-    CGNSReader,
-    DEMReader,
-    EnSightReader,
-    FacetReader,
-    FluentReader,
-    GIFReader,
-    GLTFReader,
-    HDFReader,
-    HDRReader,
-    JPEGReader,
-    MetaImageReader,
-    MFIXReader,
-    MultiBlockPlot3DReader,
-    NIFTIReader,
-    NRRDReader,
-    OBJReader,
-    OpenFOAMReader,
-    POpenFOAMReader,
-    Plot3DFunctionEnum,
-    Plot3DMetaReader,
-    PLYReader,
-    PNGReader,
-    PNMReader,
-    PointCellDataSelection,
-    PTSReader,
-    PVDDataSet,
-    PVDReader,
-    SegYReader,
-    SLCReader,
-    STLReader,
-    TecplotReader,
-    TIFFReader,
-    TimeReader,
-    VTKDataSetReader,
-    VTKPDataSetReader,
-    XMLImageDataReader,
-    XMLPImageDataReader,
-    XMLRectilinearGridReader,
-    XMLPRectilinearGridReader,
-    XMLUnstructuredGridReader,
-    XMLPUnstructuredGridReader,
-    XMLPolyDataReader,
-    XMLStructuredGridReader,
-    XMLMultiBlockDataReader,
-    DICOMReader,
-    XdmfReader,
-)
+"""Deprecated utilities module."""
+import importlib
+import inspect
+import warnings
+
+from pyvista.core.errors import DeprecationError, PyVistaDeprecationWarning
+
+
+def __dir__():
+    raise DeprecationError('The `pyvista.utilities` module has been deprecated.')
+
+
+def __getattr__(name):
+    # TODO: for some reason this runs twice
```

Comment:
What runs twice and how? Are you saying that the utilities module gets executed twice? And is it twice per `__getattr__()` call? Or twice overall, instead of the one that we'd expect due to module caching?

Reply:
It seems to be twice per `__getattr__()` call

Reply:
Can you explain in what situation you see this run twice? I can't trigger that in a quick test. But I do see another issue with two exception contexts, will post a suggestion there separately.

Reply:
aha, its not executing twice. What's happening is warnings are being thrown when `__path__` is imported which is done behind the scenes by Python. I need to make sure things like `__path__` and `__file__` are actually resolving correctly

Reply:
This is proving to give me a headache... this `__getattr__` override may not be the right approach. Will come back to this

Reply:
Can you show a use case that's giving you trouble? This seems to work in a casual test, but there might be cases I'm forgetting:
```py
def __getattr__(name):
    if name.startswith('__') and name.endswith('__'):
        # don't proxy dunder names
        raise AttributeError(f'Refusing to proxy {name!r}.')
```
We could probably relax the check and refuse to expose anything that starts with a single underscore.

Reply:
This will be addressed in a follow up PR